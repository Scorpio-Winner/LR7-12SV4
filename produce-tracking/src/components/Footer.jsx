import React from 'react';
import { Typography, Link, Box } from '@mui/material';

const Footer = () => {
  return (
    <Box component="footer" sx={{ backgroundColor: '#f5f5f5', p: 2, textAlign: 'center', width: '100%' }}>
      <Link
        href="https://github.com/Scorpio-Winner"
        target="_blank"
        rel="noopener noreferrer"
        underline="none"
        color="inherit"
        sx={{ mr: 1 }}
      >
        GitHub
      </Link>
      <Typography variant="body2" color="textSecondary" sx={{ marginRight: 0 }}>
        &copy; 2023
      </Typography>
    </Box>
  );
};

export default Footer;