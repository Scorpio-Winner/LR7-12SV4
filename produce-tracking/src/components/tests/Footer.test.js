import { render, screen } from '@testing-library/react';
import Footer from '../Footer';


test('get GitHub link', () => {
    const { getByText } = render(<Footer />);
    const linkElement = getByText(/GitHub/i);
    expect(linkElement).toBeTruthy();
  });

  test('get current year', () => {
    const { getByText } = render(<Footer />);
    const yearElement = getByText(/2023/);
    expect(yearElement).toBeTruthy();
  });

  test('get GitHub link', () => {
    const { getByText } = render(<Footer />);
    const linkElement = getByText(/GitHub/i);
    expect(linkElement).toMatchSnapshot();
  });

  test('get current year', () => {
    const { getByText } = render(<Footer />);
    const yearElement = getByText(/2023/);
    expect(yearElement).toMatchSnapshot();
  });

