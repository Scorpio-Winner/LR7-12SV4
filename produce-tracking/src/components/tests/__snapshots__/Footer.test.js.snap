// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get GitHub link 1`] = `
<a
  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9su0rk-MuiTypography-root-MuiLink-root"
  href="https://github.com/Scorpio-Winner"
  rel="noopener noreferrer"
  target="_blank"
>
  GitHub
</a>
`;

exports[`get current year 1`] = `
<p
  class="MuiTypography-root MuiTypography-body2 css-1k9vj65-MuiTypography-root"
>
  Â© 2023
</p>
`;
